static/src/javascripts/projects/common/modules/navigation/membership.js

```diff
@@ -51,15 +51,22 @@ const showAccountDataUpdateWarningMessage = accountDataUpdateWarningLink => {
     );
 };
 
-const initMembership = (): void => {
+const canShow = () => new Promise(_ => _(accountDataUpdateWarning() !== null));
+
+const show = () => {
     const updateLink = accountDataUpdateWarning();
```

Comment:
You shouldn't need to run this conditional check on `updateLink` in `show` now.

`show` will only ever be called if `canShow` has resolved with `true`, which will only happen if `accountDataUpdateWarning()` is truthy.

Reply:
True. But this makes flow happy.

Reply:
I see, in which case you could define `let updateLink` in the scope shared by `canShow` and `show` and assign a value to it in `canShow` and still keep this check in `show`?

Reply:
ah this could be a module scoped thing, so you could run `accountDataUpdateWarning()` at in the root scope and store it in a const from both `canShow` and `show`

Reply:
`let`?